Source/WebKit/UIProcess/Downloads/DownloadProxy.cpp

```diff
@@ -81,10 +81,13 @@ static RefPtr<API::Data> createData(std::span<const uint8_t> data)
 void DownloadProxy::cancel(CompletionHandler<void(API::Data*)>&& completionHandler)
 {
     if (m_dataStore) {
-        protectedDataStore()->protectedNetworkProcess()->sendWithAsyncReply(Messages::NetworkProcess::CancelDownload(m_downloadID), [this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)] (std::span<const uint8_t> resumeData) mutable {
-            m_legacyResumeData = createData(resumeData);
-            completionHandler(m_legacyResumeData.get());
-            m_downloadProxyMap->downloadFinished(*this);
+        protectedDataStore()->protectedNetworkProcess()->sendWithAsyncReply(Messages::NetworkProcess::CancelDownload(m_downloadID), [weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler)] (std::span<const uint8_t> resumeData) mutable {
+            RefPtr protectedThis = weakThis.get();
+            if (!protectedThis)
+                return;
```

Comment:
It is not OK to not call the completionhandler, it will hit an assertion in debug.

Reply:
whoops, will fix